src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -425,27 +429,82 @@ generate_key_string(const std::string& argument_name, xrt_core::patcher::buf_typ
   return argument_name + buf_string;
 }
 
+
+// Basic Itanium ABI type decoding. get_demangled_type() is referenced from ChatGPT response
 static std::string
-demangle(const std::string& mangled_name)
+get_demangled_type(const std::string& s, size_t& idx)
```

Comment:
Why does this function need `s` and `idx`.?  Can't it just take `char c`.  It feels wrong that this function updates `idx`.  I imagine caller should do `get_mangled_type(name[idx++])`